aptos-move/framework/aptos-framework/sources/transaction_validation.move

```diff
@@ -155,6 +180,14 @@ module aptos_framework::transaction_validation {
             is_simulation,
             gas_payer_address
         )) {
+            assert!(
+                permissioned_signer::check_permission_consume(
```

Comment:
since all transactions are independent - couldn't you check here value without consumption - and then consume below? that would remove the need to compute max_transaction_fee below?